server/src/main/java/org/opensearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -145,10 +147,29 @@ public class BalancedShardsAllocator implements ShardsAllocator {
         Property.NodeScope
     );
 
+    public static final Setting<Boolean> PREFER_PRIMARY_SHARD_REBALANCE = Setting.boolSetting(
+        "cluster.routing.allocation.rebalance.primary.enable",
+        false,
```

Comment:
Why is the default false?

Reply:
We do not want to enable the primary shard rebalancing by default for now. later when we gain confidence, we can enable it for remote store clusters.